src/main/java/seedu/address/model/commons/NameContainsKeywordsPredicate.java

```diff
@@ -6,28 +6,31 @@
 import seedu.address.commons.util.StringUtil;
 import seedu.address.model.travelplan.TravelPlan;
 
+
 /**
  * Tests that a {@code TravelPlanObject}'s or {@code TravelPlan}'s {@code Name} matches any of the keywords given.
  */
-public class NameContainsKeywordsPredicate implements Predicate<Object> {
+public class NameContainsKeywordsPredicate implements Predicate<TravelPlan> {
     private final List<String> keywords;
 
     public NameContainsKeywordsPredicate(List<String> keywords) {
         this.keywords = keywords;
     }
 
     @Override
-    public boolean test(Object obj) {
-        if (obj instanceof TravelPlan) {
-            return keywords.stream()
-                    .anyMatch(keyword -> StringUtil.containsWordIgnoreCase(((TravelPlan) obj).getName().name, keyword));
-        } else if (obj instanceof TravelPlanObject) {
-            return keywords.stream()
-                    .anyMatch(keyword ->
-                            StringUtil.containsWordIgnoreCase(((TravelPlanObject) obj).getName().name, keyword));
-        } else {
-            return false;
-        }
+    public boolean test(TravelPlan tp) {
+        return keywords.stream()
+                .anyMatch(keyword -> StringUtil.containsWordIgnoreCase(tp.getName().name, keyword));
+    }
+
+    /**
+     * Overloaded method to account for travel plan object.
+     * @param travelPlanObject object to be tested.
+     * @return true or false.
+     */
+    public boolean test(TravelPlanObject travelPlanObject) {
```

Comment:
Oh I see what you did there. Works but I feel like it is not following some OOP shit or smth. But I think we come back to this later ba.